### PR TITLE
Show the relevant event if present when searching by event id

### LINKED ISSUE #4283: Search by event id is broken
Searching by event id no longer shows you the event details. Instead it opens the error group with the latest event showing. This is problematic because there's no other way to get info about the exact event.

### PR DESCRIPTION
**Pull request by JamesMura**
This pull request fixes #4283

### CODE REVIEW

File: src/sentry/api/endpoints/project_group_index.py [hunk 1]
```diff
@@ -226,6 +228,10 @@ def get(self, request, project):
                     pass
                 else:
                     matching_group = Group.objects.get(id=mapping.group_id)
+                    try:
+                        matching_event = Event.objects.get(event_id=query)
```
Comment (mattrobenolt) [inline src/sentry/api/endpoints/project_group_index.py]: This is incorrect and could be a security hole.

`event_id`'s aren't globally unique, so this query right now is making the assumption that it is, and we will only find one row in the entire database to match.

If we look at our model definition, https://github.com/getsentry/sentry/blob/master/src/sentry/models/event.py#L57

We are unique'd on `project_id, event_id`. This is just saying that `event_id`'s are unique per project scope.

Therefore, we also need to qualify the query here with `project_id`

So:

```python
Event.objects.get(
  project_id=project.id,
  event_id=query,
)
```

Would give us the results we expect.

**Review by mattrobenolt — CHANGES_REQUESTED** [review]
Other than the change to the `Event` query, this seems reasonable.

File: src/sentry/api/endpoints/project_group_index.py (hunk 1 above)
Comment (JamesMura) [inline src/sentry/api/endpoints/project_group_index.py]: makes sense, will fix that

**Review by mattrobenolt — APPROVED** [review]
🥇

**Pull request merged**